src/context/HicetnuncContext.js

```diff
@@ -609,9 +590,9 @@ class HicetnuncContextProviderClass extends Component {
           .catch((e) => e)
       },
 
-      cancel: async (swap_id) => {
+      cancel: async (contract_address, swap_id) => {
         return await Tezos.wallet
-          .at(this.state.proxyAddress || this.state.v2)
+          .at(this.state.proxyAddress || contract_address)
```

Comment:
Here, when collab contract (proxy) is equipped, contract call would fail, because proxy would try to cancel swap in v2 marketpace (which is hardcoded in collab contracts). In the collab contract equipped case it is required to make call to the `execute` entrypoint of the collab with lambda and params